src/coverlet.core/Reporters/CoberturaReporter.cs

```diff
@@ -129,5 +132,37 @@ public string Report(CoverageResult result)
 
             return Encoding.UTF8.GetString(stream.ToArray());
         }
+
+        private static IEnumerable<string> GetRootDirs(Modules modules, bool isSourceLinkUsed)
+        {
+            if (isSourceLinkUsed) return new[] { string.Empty };
+
+            List<string> sources = new List<string>();
+
+            foreach (var module in modules)
+            {
+
+                sources.AddRange(
+                    module.Value.Select(d => Path.GetDirectoryName(d.Key)));
+            }
+
+            sources = sources.Distinct().ToList();
+            return sources.Select(Directory.GetDirectoryRoot).Distinct();
+        }
+
+        private static string GetRelativePathFromBase(IEnumerable<string> basePaths, string path, bool isSourceLinkUsed)
+        {
+            if (isSourceLinkUsed) return path;
+
+            string relativePath = path;
+            basePaths.ToList().ForEach(basePath =>
+            {
+                if (path.StartsWith(basePath))
+                {
+                    relativePath = path.Substring(basePath.Length);
```

Comment:
what do you think about
```cs
            if (isSourceLinkUsed)
            {
                return path;
            }

            foreach (var root in basePaths)
            {
                if (path.StartsWith(root))
                {
                    return path.Substring(root.Length);
                }
            }

            Debug.Assert(false, "Unexpected, we should find at least one path starts with one pre-build root");

            return path;
```
it seems more logic to me...when we found one valid root we can return immediatly, if we don't found any valid root(unexpected) we return original path, so we avoid exception and we'll generate a file(better than nothing), I added also an assertion so on debug we'll catch possible problem.

Reply:
Yes, I agree.